### PR TITLE
Finalised

### PR DESCRIPTION
**Pull request by YouQuan2007**
I have made
1. Add "Contact Us" button in both student dashboard and lecturer dashboard, this is to avoid the switching scenes issue from contact us to course registration when the name is not saved
2. Whenever a user try to switch to course registration, an error message will occur. Then, if the student save all the info except name, an error message will also appear.
3. Update the latest version of loginController after double check with Michelle's latest version
For CGPA validation and name saving I have passed to Michelle

### CODE REVIEW

File: data/Student Dashboard/ASD.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+2233
+Active
+18/1
+3.44
+Year 2
+School of Management
+Main Campus
+BSc Management
+Software Engineering
+Mathematics
```
Comment (Michelle-Lohwt) [inline data/Student Dashboard/ASD.txt]: Delete this file

File: data/Student Dashboard/XXX.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+3232
+Active
+18/2
+3.65
+Year 1
+School of Computer Science
+Main Campus
+BSc Computer Science
+Intelligent Computing
+Electives
```
Comment (Michelle-Lohwt) [inline data/Student Dashboard/XXX.txt]: Delete this file

File: data/Student Dashboard/YYQ.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+010101
+Active
+18/2
+3.55
+Year 2
+School of Mathematical Sciences
+Main Campus
+BSc Mathematics
+Intelligent Computing
+Accounting
```
Comment (Michelle-Lohwt) [inline data/Student Dashboard/YYQ.txt]: Delete this file

File: data/Student Profile/123456.txt [hunk 4]
```diff
@@ -1,2 +0,0 @@
-123456
-123456
```
Comment (Michelle-Lohwt) [inline data/Student Profile/123456.txt:2]: delete this file or restore the information in this file

File: data/Student Profile/999000.txt [hunk 5]
```diff
@@ -0,0 +1,2 @@
+999000
+111111
```
Comment (Michelle-Lohwt) [inline data/Student Profile/999000.txt]: Delete this file

File: src/sample/CallUs.fxml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+            <Font size="14.0" />
+         </font>
+      </Text>
+      <Text layoutX="51.0" layoutY="148.0" strokeType="OUTSIDE" strokeWidth="0.0" text="Phone No. : +604-653 3888" wrappingWidth="236.53675842285156">
+         <font>
+            <Font size="14.0" />
+         </font>
+      </Text>
+   </children>
+</AnchorPane>
```
Comment (Michelle-Lohwt) [inline src/sample/CallUs.fxml]: Put up a pic

File: src/sample/lecDash.fxml [hunk 7]
```diff
@@ -209,9 +209,19 @@
...
+                  <Text fx:id="errorMessage" fill="#fa0000" layoutX="318.0" layoutY="57.0" strokeType="OUTSIDE" strokeWidth="0.0" textAlignment="CENTER" wrappingWidth="220.800048828125">
+                     <font>
+                        <Font name="System Bold" size="18.0" />
+                     </font>
+                  </Text>
+                  <JFXButton fx:id="contactButton" layoutX="690.0" layoutY="448.0" onAction="#CallUs" prefHeight="40.0" prefWidth="170.0" style="-fx-background-color: #EB7B31; -fx-cursor: hand;" text="Contact Us" textFill="WHITE">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </JFXButton>
```
Comment (Michelle-Lohwt) [inline src/sample/lecDash.fxml]: Put up the pic

File: src/sample/lecDashController.java [hunk 8]
```diff
@@ -40,13 +57,38 @@ public void ContactUs(MouseEvent event) throws IOException {
...
+      Image icon = new Image("sample/images/usm-ringlogo.png");
+      stage.getIcons().add(icon);
+      stage.setTitle("Contact Us");
+      stage.setResizable(false);
+      stage.setScene(new Scene(root1));
+      stage.show();
+    } catch (Exception e) {
+      System.out.println("Can't load new window");
+    }
+  }
```
Comment (Michelle-Lohwt) [inline src/sample/lecDashController.java:74]: Put a video about this

File: src/sample/lecDashController.java [hunk 9]
```diff
@@ -89,8 +249,10 @@ private void ChoiceBoxItem() {
 
   @Override
   public void initialize(URL location, ResourceBundle resources) {
+    staffID.setText(id);
     ChoiceBoxItem();
     defaultInfo();
+
```
Comment (Michelle-Lohwt) [inline src/sample/lecDashController.java]: remove extra line

File: src/sample/stuDash.fxml [hunk 10]
```diff
@@ -258,9 +258,19 @@
...
+                  <Text fx:id="Messages" fill="#f70000" layoutX="273.0" layoutY="63.0" strokeType="OUTSIDE" strokeWidth="0.0" textAlignment="CENTER" wrappingWidth="384.0">
+                     <font>
+                        <Font name="System Bold" size="18.0" />
+                     </font>
+                  </Text>
+                  <JFXButton fx:id="contactButton" layoutX="690.0" layoutY="448.0" onAction="#CallUs" prefHeight="40.0" prefWidth="170.0" style="-fx-background-color: #EB7E31; -fx-cursor: hand;" text="Contact Us" textFill="WHITE" AnchorPane.leftAnchor="690.0" AnchorPane.topAnchor="448.0">
+                     <font>
+                        <Font size="18.0" />
+                     </font>
+                  </JFXButton>
```
Comment (Michelle-Lohwt) [inline src/sample/stuDash.fxml]: Put a pic about this

File: src/sample/stuDashController.java [hunk 11]
```diff
@@ -72,29 +136,236 @@ public void editStuInfo() {
...
-    school.setDisable(true);
-    campus.setDisable(true);
-    programme.setDisable(true);
-    major.setDisable(true);
-    minor.setDisable(true);
+    //If want to do CGPA validation then have to convert string which is cgpa.getText() to int,
+    // and this is the way I found online, have tried several times
+    //but cannot
+    /**String cgpaInteger=cgpa.getText();
+    int cgpaNumber = Integer.valueOf(cgpaInteger);
```
Comment (Michelle-Lohwt) [inline src/sample/stuDashController.java]: How can you input a string as float and request an integer

File: src/sample/stuDashController.java [hunk 12]
```diff
@@ -105,11 +376,20 @@ private void ChoiceBoxItem() {
     programme.getItems().addAll("BSc Computer Science", "BSc Mathematics", "BSc Management");
     major.getItems().addAll("Software Engineering", "Intelligent Computing", "Computing Infrastructure");
     minor.getItems().addAll("Accounting", "Mathematics", "Electives");
+    
+
```
Comment (Michelle-Lohwt) [inline src/sample/stuDashController.java]: remove extra lines

**Review by Michelle-Lohwt — CHANGES_REQUESTED** [review]
Check the comments

File: src/sample/lecDashController.java (hunk 8 above)
Comment (YouQuan2007) [reply src/sample/lecDashController.java:74]: https://user-images.githubusercontent.com/93238154/151544125-62117047-edc6-4028-b41c-6af3a30cca3d.mp4

File: src/sample/stuDash.fxml (hunk 10 above)
Comment (GitHubUser5136) [reply src/sample/stuDash.fxml]: Actually @YouQuan2007 , you can try to pass the name variable from studash.java to contactus.java. If the name variable is empty then cannot switch scene from contact us to courseregcontroller. There is no need to create another fxml (CallUs.fxml)

File: src/sample/lecDash.fxml (hunk 7 above)
Comment (GitHubUser5136) [reply src/sample/lecDash.fxml]: Actually @YouQuan2007  , you can try to pass the name variable from studash.java to contactus.java. If the name variable is empty then cannot switch scene from contact us to courseregcontroller. There is no need to create another fxml (CallUs.fxml)

**Review by GitHubUser5136 — CHANGES_REQUESTED** [review]

File: src/sample/CallUs.fxml (hunk 6 above)
Comment (YouQuan2007) [reply src/sample/CallUs.fxml]: photo_2022-01-28_20-16-05

File: src/sample/lecDash.fxml (hunk 7 above)
Comment (YouQuan2007) [reply src/sample/lecDash.fxml]: photo_2022-01-28_20-16-05

**Pull request closed without merging**